### ISSUE TITLE
Should we first shuffle the dataset then batch it?

### ISSUE THREAD

**Issue by zihaozhihao** [HasnainRaz/SemSegPipeline#5]
https://github.com/HasnainRaz/Tensorflow-input-pipeline/blob/819e8456e89a61b6c40c2f49bb3306bc2c2a2d87/dataloader.py#L211

**Comment by HasnainRaz** [comment]
Uh yeah, that would be better, I'll push a fix soon, thanks for pointing it out.
